### ISSUE TITLE
GREAT WORK

### ISSUE THREAD

**Issue by bbombek** [virgilm/raspipool#28]
So I have struggled with this build for one year. It seems to be very hard....
And HA is constantly changing. Now the new version doesnt support UART?? Is that a problem?
Anyway i am again stuck.
I see that
![image](https://user-images.githubusercontent.com/82860165/152689499-765bbe7b-016f-4ea3-bbcd-f21c969f5eb0.png)

Also the readme i got stuck at

![image](https://user-images.githubusercontent.com/82860165/152689541-4f89b5f8-1d3a-499e-9eec-41c416d616c7.png)

I dont know how how to do no. 6 and 7.
and also i got now two red nodes??

![image](https://user-images.githubusercontent.com/82860165/152689602-30835aa8-c310-4ab9-badc-c5dae3a887ff.png)

And I still dont get any readings from EZo circuits. I Dont know how to enable i2C mode....
and also
what is this?
![image](https://user-images.githubusercontent.com/82860165/152689665-adde741c-410e-4d01-acd2-55cb649ac2bf.png)

what is sparkly local i dont understans??
PLEASE HELP

**Comment by virgilm** [comment]
To set EZO sensors to I2C mode, you can follow [this](https://www.instructables.com/UART-AND-I2C-MODE-SWITCHING-FOR-ATLAS-SCIENTIFIC-E/) tutorial. My version does not support UART mode, you would have to make it work if you really wanted to (the code is there, but commented out). I highly recommend using I2C mode though, it is easier to use and more reliable.

HA is indeed constantly changing (and not for the better IMO). I assume you installed the supervised version of HA, seeing the screenshots - and that is OK. Steps 6 & 7 are irrelevant to you, NR is running and connected to HA apparently, and the second NR entry is useless - just ignore it or remove it from your configuration file.

As I mentioned in the readme, the installation is very much a work in progress and I use HA Core, where some if this stuff is more complicated - but it runs much better on my RPi Zero W 2. I'm actually happy you made it that far, if you can contribute back with Install steps for HA Supervised, that would be nice.

Finally, my installation is called sparkly.local instead of homeassistant.local, so you already have that last point (Start HA and access) working :)

**Comment by bbombek** [comment]
OK KUULL
Getting there!
So I enabled I2C mode manually on the EZO circuits now there is the Blue Light!
But still no communication with raspberry pi.
I know how to do it in raspbian OS
https://files.atlas-scientific.com/pi_sample_code.pdf
But how to do that in HA OS??
I found this should i do it?
https://www.home-assistant.io/common-tasks/os/#enable-i2c
I will gladly make a tutorial if I am Successful!

**Comment by bbombek** [comment]
So I did this also..
https://www.home-assistant.io/common-tasks/os/#enable-i2c-via-home-assistant-operating-system-terminal
And nothing from the pH sensor...
![image](https://user-images.githubusercontent.com/82860165/152698767-a4730805-9663-4b4b-88b4-2601d40e59b0.png)


Should sensors still be connected as segalion did??
![image](https://user-images.githubusercontent.com/82860165/152698743-4e82a130-0ea9-46ad-a6b2-36fcfdf1f5ef.png)

**Comment by virgilm** [comment]
Yeah, that won't work with I2C. So I guess you have a choice to make:
- either figure out how to re-enable serial in sparkly (you should be able to by changing the ph.yaml and orp.yaml (the beginning of the file).
- or figure out how to connect your sensors via I2C. Not sure if that carrier board in the schematic works in I2C mode, I am using the tentacle T3 carrier board. (edit: the [carrier board](https://files.atlas-scientific.com/Isccb.pdf) seems to work in I2C mode too).
If you decide to go the I2C route, [this](https://community.home-assistant.io/t/add-on-hassos-i2c-configurator/264167) helped me get I2C running on HA, which makes even the most trivial tasks complicated.

**Comment by askewdread** [comment]
awesome updates to the code,



> 
> Should sensors still be connected as segalion did?? ![image](https://user-images.githubusercontent.com/82860165/152698743-4e82a130-0ea9-46ad-a6b2-36fcfdf1f5ef.png)

have finally got this working after much trial and error, the sensor boards need to be connected as per the pi_sample.pdf file not as per the image in the repo, so RX on both boards need connecting to the SCL pin on the PI and the TX on both boards need to be connected to the SDA pin on the PI

then HA correctly finds them and reads :)

**Comment by bbombek** [comment]
Hello I am Back!
Now i finally got the Whitebox T3 carrier board for RPI. Now still nothing in HA no readings from sensors...
What to do. Can someone write a quick tutorial.??
THANK YOU

**Comment by STELTZ-MANIA** [comment]
Hi @bbombek, how did you get on or have you quit?

I received my Tentacle T3 today and have hooked up the EZO pH and ORP circuits and it appears to be working. 
![image](https://user-images.githubusercontent.com/103625353/188543554-bcd8662e-7389-4261-8d4b-d2d8f37cdf05.png)

Although there is a weird difference between the 2 pH values being presented which I couldn't be bothered thinking about right now and I haven't added any temperature sensors as yet.

I'm happy for us to compare notes if you are still having issues.
